CryptoLib/CryptoLib/Decrypt.mm

```diff
@@ -23,65 +23,50 @@
 #import "Decrypt.h"
 #import "cdoc/CdocReader.h"
 #import "cdoc/Token.h"
-#import "ApduToken.h"
-#import "XmlParserDelegate.h"
+#import "SmartCardTokenWrapper.h"
+#import "DdocParserDelegate.h"
 #import "base64.h"
 #import <UIKit/UIKit.h>
 
 @implementation Decrypt
 
-- (BOOL)decryptFile: (NSString *)fullPath withPin :(NSString *) pin withToken :(ApduToken *) apduToken {
+- (NSMutableDictionary *)decryptFile: (NSString *)fullPath withPin :(NSString *) pin withToken :(AbstractSmartToken *) smartToken {
     
     std::string encodedFullPath = std::string([fullPath UTF8String]);
     std::string encodedPin = std::string([pin UTF8String]);
     CDOCReader cdocReader(encodedFullPath);
-    
+    SmartCardTokenWrapper * apduToken = new SmartCardTokenWrapper(encodedPin, smartToken);
```

Comment:
There is no `delete` called on the memory object owned by `apduToken` object, thus creates memory leak.
Smart pointer should be used instead of `new`, preferably `unique_ptr<>`.

If `delete` is called somewhere in the called method definitions it is bad approach also.